datumaro/plugins/yolo_format/extractor.py

```diff
@@ -2,45 +2,61 @@
 #
 # SPDX-License-Identifier: MIT
 
+from __future__ import annotations
+
 import os.path as osp
 import re
 from collections import OrderedDict
-from typing import Dict
-
-from datumaro.components.annotation import AnnotationType, Bbox, LabelCategories
+from typing import Dict, List, Optional, Tuple, Type, TypeVar, Union
+
+from datumaro.components.annotation import Annotation, AnnotationType, Bbox, LabelCategories
+from datumaro.components.errors import (
+    DatasetImportError,
+    InvalidAnnotationError,
+    UndeclaredLabelError,
+)
 from datumaro.components.extractor import DatasetItem, Extractor, Importer, SourceExtractor
 from datumaro.components.format_detection import FormatDetectionContext
 from datumaro.components.media import Image
-from datumaro.util.image import DEFAULT_IMAGE_META_FILE_NAME, load_image_meta_file
+from datumaro.util.image import DEFAULT_IMAGE_META_FILE_NAME, ImageMeta, load_image_meta_file
 from datumaro.util.meta_file_util import has_meta_file, parse_meta_file
 from datumaro.util.os_util import split_path
 
 from .format import YoloPath
 
+T = TypeVar("T")
+
 
 class YoloExtractor(SourceExtractor):
     class Subset(Extractor):
-        def __init__(self, name, parent):
+        def __init__(self, name: str, parent: YoloExtractor):
             super().__init__()
             self._name = name
             self._parent = parent
-            self.items = OrderedDict()
+            self.items: Dict[str, Union[str, DatasetItem]] = OrderedDict()
 
         def __iter__(self):
             for item_id in self.items:
-                yield self._parent._get(item_id, self._name)
+                item = self._parent._get(item_id, self._name)
+                if item is not None:
+                    yield item
```

Comment:
This enables questionable semantics: if any items are skipped in this way, then the number of items yielded will be different from the value returned by `len(subset)`.

Reply:
IMO, the proper solution would involve removing `__len__` from `IExtractor`.